internal/factsengine/gatherers/ascsers_cluster_test.go

```diff
@@ -286,3 +287,26 @@ func (suite *AscsErsClusterTestSuite) TestAscsErsClusterGather() {
 	}
 	suite.ElementsMatch(expectedEntries, entries)
 }
+
+func (suite *AscsErsClusterTestSuite) TestAscsErskGathererContextCancelled() {
+	ctx, cancel := context.WithCancel(context.Background())
+	cancel()
+
+	suite.mockExecutor.
+		On("ExecContext", mock.Anything, "cibadmin", "--query", "--local").
```

Comment:
Did you try putting `ctx` instead of `mock.Anything`?
Maybe it validates that the function is called with the expected argument.
I don't know if it works though.

Reply:
It works and it's actually brilliant! I refactored all tests like that in https://github.com/trento-project/agent/pull/379/commits/596152435a42cc5abe3aaf86f435937ca1e89013

It would have spotted the bug in https://github.com/trento-project/agent/pull/379#discussion_r1926499834
You made my day